### DISCUSSION TITLE
I want to implement AD group based kyuubi engine, can we have multiple engine types for 1 kyuubi deployment?

### DISCUSSION THREAD

**Discussion by vijay-jangir** [apache/kyuubi#6847] [Q&A]
I want 2 different kyuubi engines, 1. Spark engine, 2. Trino engine. 

For user belonging to X group, spark engine should spawn at group level isolation
for users belonging to Y group, trino engine should spawn at user level isolation.

Can i achieve this?

**Comment by pan3793** [comment]
you may need to implement a custom `SessionConfAdvisor` to inject the configurations to control the session's share level and engine type
https://kyuubi.readthedocs.io/en/master/extensions/server/configuration.html
